### PR TITLE
Fixes #30799 - Upgrade js tests to support Jest 26

### PR DESCRIPTION
**Pull request by johnpmitsch**
Related to https://github.com/theforeman/foreman-js/pull/212

### CODE REVIEW

**Comment by ares** [comment]
@amirfefer it seems this will have to be merged together with the linked PR, that you're assgined to. Is it OK to assign you to as a reviewer for this one too?

File: webpack/assets/javascripts/react_app/components/common/Slot/__snapshots__/Slot.test.js.snap
Comment (johnpmitsch) [inline webpack/assets/javascripts/react_app/components/common/Slot/__snapshots__/Slot.test.js.snap:206]: I think there is some sort of unique name issue that jest now enforces, i.e. you can no longer have `Slot-Fill render multiple fills 1` and `Slot-Fill render multiple fills 2`

**Comment by johnpmitsch** [comment]
@sharvit @amirfefer Got everything passing :green_circle: 
```
Test Suites: 240 passed, 240 total
Tests:       1 skipped, 935 passed, 936 total
Snapshots:   672 passed, 672 total
Time:        157.809 s
Ran all test suites.
```

to test, please use https://github.com/theforeman/foreman-js/pull/212 with the instructions here https://github.com/theforeman/foreman-js/blob/master/CONTRIBUTING.md#testing-pull-requests

**Comment by ezr-ondrej** [comment]
This is blocked by Realease of npm @theforeman/test - as it is major change (5.0.1 or 5.1 should get released)

File: package.json
Comment (johnpmitsch) [inline package.json:35]: @sharvit should I just update all of the `@theforeman` packages to 6.0.0 or just the test package?

Comment (ezr-ondrej) [reply package.json:35]: well, you depend only on the `test` right? so I'd say bump only the required package and if that needs newer versions of the others, right?

Comment (ezr-ondrej) [reply package.json:35]: And, you probably do not need packaging PR if you don't bump the others if I'm not mistaken ...

**Review by ezr-ondrej — DISMISSED** [review]

**Comment by ezr-ondrej** [comment]
Needs packaging ack.

File: package.json
Comment (johnpmitsch) [reply package.json:35]: @theforeman/packaging any packaging PR needed for just the bump in test package?

Comment (sharvit) [reply package.json:35]: We should use the same versions for all @theforeman/* packages for comparability (they require each other).
Anyway, only the `test` package contains actual changes.

**Review by ekohl — DISMISSED** [review]
Packaging wise this is ignored so :+1: on that.

File: package.json
Comment (tbrisker) [inline package.json:24]: Would it make sense to pin this to the exact version we have in packaging to ensure dev and production are on the same version?

File: package.json
Comment (tbrisker) [inline package.json]: This one should also be bumped

File: package.json
Comment (johnpmitsch) [reply package.json:24]: Maybe it would make sense to switch to `~` so it would update to the latest z version but not update the y version until we manually updated.

so `~6.0.0` would pull in 6.0.1, 6.0.2, etc... but not 6.1.0

@sharvit  thoughts?

Also please double check my understanding of tilde and caret ranges here https://docs.npmjs.com/cli/v6/using-npm/semver :)

File: package.json
Comment (johnpmitsch) [reply package.json]: thanks!

**Comment by johnpmitsch** [comment]
The only major breaking change is the jest upgrade (we skipped version 5.0.0 so no changes there), so I don't think any additional code changes are needed, but feel free to point out anything I'm missing :) I'll open up a packaging PR as well.

**Comment by johnpmitsch** [comment]
opened https://github.com/theforeman/foreman-packaging/pull/5979

File: package.json
Comment (sharvit) [reply package.json:24]: I remember a long discussion where we decided to keep the `^` so we can catch errors in the development environment faster.

**Review by sharvit — DISMISSED** [review]
Thanks @johnpmitsch

**Review by ezr-ondrej — DISMISSED** [review]
:+1:

File: package.json
Comment (tbrisker) [reply package.json:24]: Any chance you can find that discussion @sharvit ? i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version, so we won't know something is broken until the packages build in the best case or until someone notices something doesn't work in the worst case. 
For example - https://github.com/theforeman/foreman/pull/7505#pullrequestreview-523158457 - tests were green but merging would cause nightly to break and there was no indication that a version bump was needed.

File: package.json
Comment (ekohl) [reply package.json:35]: What probably should happen is that any time you depend on a feature introduced in 6.0.1, you actually update the dependency to 6.0.1. In practice, people often forget this.

File: package.json
Comment (johnpmitsch) [reply package.json:24]: Not sure if it's the exact convo, but I know I brought it up here: https://community.theforeman.org/t/pinning-npm-packages-to-y-version/12393

Note that this is before we had the foreman-js meta-package separation, so it could make more sense now to use strict pinning of versions for `@theforeman/*` packages. That way it forces us to update foreman and foreman-packaging to use updated packages in y and z version releases of the foreman-js packages.

Comment (sharvit) [reply package.json:24]: > Any chance you can find that discussion @sharvit ?

I tried but couldn't find any.

>  i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version

>  i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version, so we won't know something is broken until the packages build in the best case or until someone notices something doesn't work in the worst case.

Does it make a difference because it's `@theforeman/*` package or you would suggest that to everything in the `package.json`?

> For example - #7505 (review) - tests were green but merging would cause nightly to break and there was no indication that a version bump was needed.

I am not sure it's a good example because we merge the `helmet` upgrade after we made sure no one using that feature yet and we discussed with the PR author before and with other UI maintainers before merging a releasing as a none breaking changes.
If it was not the case, we would release it as a major release.

**Comment by johnpmitsch** [comment]
Is this good to merge or should the packaging PR go in first?

**Comment by tbrisker** [comment]
I'd still prefer to pin a specific version but that doesn't have to block this PR. Waiting for packaging to be merged before merging this one.

**Comment by ezr-ondrej** [comment]
> Is this good to merge or should the packaging PR go in first?

We need to get the packaging first, so we wouldn't get this in and then forgot about the packaging...

**Review by tbrisker — APPROVED** [review]
Packaging has been merged. Thanks @johnpmitsch !

**Pull request merged**
